toolkit/scripts/toolchain/check_manifests.sh

```diff
@@ -8,19 +8,30 @@ TOOLCHAIN_SPEC_LIST_FILE=$1
 SPECS_DIR=$2
 MANIFESTS_DIR=$3
 DIST_TAG=$4
-ARCH=$5
+RELEASE_NUM=$5
+ARCH=$6
+
+# Define 'azl <ver>' macro for use in spec files. Normally they would be defined in the 'azurelinux-release' package.
+# Dist tag will be of the form '.<distro><ver>', e.g. '.azl3'., rease number will be of the form '3.0'.
+
+# 'azl <ver>': Strip the leading '.' and the trailing number from the DIST_TAG
+distro_name="$(echo ${DIST_TAG#.} | sed 's/[0-9]\+$//')"
+distro_majver="${RELEASE_NUM%%.*}"
+distro_ver="${distro_name} ${distro_majver}"
+
+distro_defines=(--define "with_check 1" --define "dist $DIST_TAG" --define "$distro_ver")
 
 write_rpms_from_spec () {
     # $1 = spec file
     # $2 = file to save to
     spec_dir=$(dirname $1)
-    exclusiveArch=$(rpmspec -q $1 --define="with_check 0" --define="_sourcedir $spec_dir" --define="dist $DIST_TAG" --qf="[%{EXCLUSIVEARCH} ]" --srpm 2>/dev/null)
+    exclusiveArch=$(rpmspec -q $1 --define="with_check 0" --define="_sourcedir $spec_dir" "${distro_defines[@]}" --qf="[%{EXCLUSIVEARCH} ]" --srpm 2>/dev/null)
```

Comment:
Now we're changing the behaviour here a bit. We're starting with `--define="with_check 0"` and then pass `--define "with_check 1"` insice `distro_defines`. I'm not sure, if anything may break but if we're making that change anyway, I'm curious, if we could switch to calling `mariner_rpmspec` altogether. This way the logic setting the defines would be kept in one place less.